<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Maggie Shaw" />


<title>CellRanger Count</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">My Website</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="bioinformatics_basics.html">Bioinformatics Basics</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Pipeline Walkthrough's
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="WT_BarcodeLibrary_MS08Oct2021.html">Walkthrough: Library/Barcode Projects</a>
    </li>
    <li>
      <a href="BarcodeBatchforLibraryProjects_MS16July2021.html">How to Run: Library/Barcode Projects</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Guides
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Archiving_Guide_26Mar2020.html">How to: Archive</a>
    </li>
    <li>
      <a href="ConcatenatingGuide.html">How to: Concatenate</a>
    </li>
    <li>
      <a href="CellRanger_Count.html">CellRanger - Count</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">CellRanger Count</h1>
<h4 class="author">Maggie Shaw</h4>
<h4 class="date">6/23/2021</h4>

</div>


<div id="how-to-use-cellranger-count-and-combine-mulitple-runs" class="section level2">
<h2>How to use CellRanger <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/using/count">Count</a> and combine mulitple runs</h2>
<p>Basic instructions for running the count tool from CellRanger with instructions on how to combine multiple runs of the same sample.</p>
<div id="basic-count-command" class="section level3">
<h3>Basic Count Command</h3>
<p><strong>Basic Non-Submitted Command</strong></p>
<pre><code>cellranger count --id=4657 
--transcriptome=/home/shawma/CellRanger/testdata_28Jan2021/reference_test/refdata-gex-mm10-2020-A --fastqs=/home/shawma/CellRanger/testdata_28Jan2021/H7JT5BGXH/outs/fastq_path/ 
--sample=4657 --include-introns --localcores=16 --localmem=100 --r1-length=26</code></pre>
<p>Breakdown </br> - For each sample change, <strong>–id=</strong> and <strong>–sample=</strong> </br> - <strong>id</strong> = name of the output file for that unique sample name </br> - <strong>sample</strong> = unique sample identifier, raw fastqs should be a folder with this sample id for each sample </br> - The <strong>–transcriptome</strong> tag should be the output from the directory from using CellRanger <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/advanced/references">mkref</a> </br> - use the <strong>–include-introns</strong> tag when running 10x on nuclei - use full path for <strong>transcriptome</strong> and <strong>fastqs</strong> </br> - <strong>-NOTE : –r1-length=26</strong> This was added after the fact, got an error that said some of our UMIs were greater than length 26 (26-28 was the range), this just forces them to 26, seemed to resolve the issue</p>
<p><strong>IMPORTANT SETUP NOTE</strong> - Only one sample’s fastqs should be in a folder. One folder for each sample. The output will follow the same format.</p>
</div>
<div id="submitting-to-the-hpc" class="section level3">
<h3>Submitting to the HPC</h3>
<p>CellRanger is preloaded on the HPC as a module. When you submit the job be sure to also load the module. The count step is computationally taxing and will take some time to complete, be sure to ask for addtional memory and nodes.</p>
<pre><code>bsub -o count_out.txt -e count_error.txt -M 100000 -n 16 &quot;module load cellranger/5.0.1;cellranger count --id=4657 --transcriptome=/home/shawma/CellRanger/testdata_28Jan2021/reference_test/refdata-gex-mm10-2020-A --fastqs=/home/shawma/CellRanger/testdata_28Jan2021/H7JT5BGXH/outs/fastq_path/ --sample=4657 --include-introns --localcores=16 --localmem=100 --r1-length=26&quot;</code></pre>
<ul>
<li>make sure the <strong>-M</strong> and <strong>-n</strong> in the job submission match <strong>–localmem</strong> and <strong>–localcores</strong>, the two cell ranger commands give a limit to the allocated local memory and cores. The job submission commands should not exceed these numbers as it may cause problems with the queue and job submission. For now its safest to match them.</li>
</ul>
</div>
<div id="submitting-multiple-runs-for-each-sample" class="section level3">
<h3>Submitting multiple runs for each sample</h3>
<p>To submit multiple runs ALL runs for a sample must have the same <strong>sampleID</strong> folder name and file prefix. They can be in different locations (ie B# folders) but the imediate directory name must be identical to eachother and to the <strong>–sample</strong> tag. CellRanger allows you to comma deliminate the <strong>–fastqs</strong> field to list multiple runs to pool together.</p>
<pre><code>bsub -o count_out.txt -e count_error.txt -M 100000 -n 16 &quot;module load cellranger/5.0.1;cellranger count --id=4657 --transcriptome=/home/shawma/CellRanger/testdata_28Jan2021/reference_test/refdata-gex-mm10-2020-A --fastqs=/home/shawma/CellRanger/testdata_28Jan2021/H7JT5BGXH/outs/fastq_pathA/,/home/shawma/CellRanger/testdata_28Jan2021/H7JT5BGXH/outs/fastq_pathB/,/home/shawma/CellRanger/testdata_28Jan2021/H7JT5BGXH/outs/fastq_pathC/ --sample=4657 --include-introns --localcores=16 --localmem=100 --r1-length=26&quot;</code></pre>
<p>To understand this a little better, for the example above you would find a set of fastqs(corresponding R1,R2 I1, I2) for the single sample 4657 in the following directories: </br> <strong>Run #1</strong> - /home/shawma/CellRanger/testdata_28Jan2021/H7JT5BGXH/outs/fastq_pathA/4657 </br> <strong>Run #2</strong> - /home/shawma/CellRanger/testdata_28Jan2021/H7JT5BGXH/outs/fastq_pathB/4657 </br> <strong>Run #3</strong> - /home/shawma/CellRanger/testdata_28Jan2021/H7JT5BGXH/outs/fastq_pathC/4657 </br></p>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
