---
title: 'Walkthrough: Library and Barcode Project Analysis'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
### Introduction
In the Library Analysis Pipeline, we use paired-end, short read sequencing and trade traditional alignment, for a method where we remove the identical regions surrounding a small variable region and tally the unique occurrences of the <span style="color:firebrick">**region of interest**</span>. After identifying and extracting the ROI, you can discover hits by calculating enrichment and characterizing the represented sequences.


![In a library or barcode study, all paired reads should have an identical structure.](images\librarypool.svg)

---

### **Input**

Rather than a map, the analysis requires the expected identity of the regions flanking the sequence of interest. With the ROI in the center, the "adapters" and ROI should make up the entire length of the amplicon.

![The 5' and 3' "Adapters" should be nearly identical across all reads. These adapters will be trimmed so that we can extract the unique library members.](images\libraryamplicon.svg)

---

### **Step 1:** Join paired-end reads

![When using paired end reads, there should be a few bases of overlapping region that can be used to properly align reads.](images\libraryR1R2.svg)

In traditional alignment, paired end reads are joined by the mapping software during the alignment process. Here we must join the paired reads using **fastq-join** a software specific to joining reads without alignment. To read more about **fastq-join** please see their [github](https://github.com/brwnj/fastq-join).

#### **Parameters Used:**

<span style="color:Purple">**-p 10**</span>   - the overlapping region can not exceed <span style="color:Purple">**10%**</span> difference in similarity
        
---

### **Step 2:** Take the reverse compliment and concatenate to foward joined reads

Using the tool <span style="color:Purple">**fastx_reverse_compliment**</span> from <span style="color:Purple">**FASTX_toolkit**</span>, the joined reads are reversed and reverse reads are added to the end of the reads list in the joined fastq file. The purpose of this is to ensure that ampljcons that were accidently reversed in seqeuncing included in the final library counts. For more information on <span style="color:Purple">**fastx_reverse_compliment**</span>, see the <span style="color:Purple">**FASTX_toolkit**</span> [webpage](http://hannonlab.cshl.edu/fastx_toolkit/commandline.html#fastx_revcomp_usage) 

**NOTE:** This step should double your joined reads file size. Reads already in the correct orientation will not be double counted from this step.

![Foward and reverse orientation of the reads are put into a single file.](images\libraryforwardreverse.svg)
---

### **Step 3:** Trim "adapters"

**Cutadapt** is a tool used to *"find and remove adapter sequences, primers, poly-A tails and other types of unwanted sequence from your high-throughput sequencing reads"*. Here the flanking sequences we call "adapters", aren't adapters by the standard definition. We call them "adapters" to indicate that they are unwanted sequences and should be trimmed before analysis.

The expected identity of the sequences are used to mark what should be trimmed and unmatched reads are discarded. The trimming of adapters is error tolerant, allowing for mismatch between the expected adapters and actual reads. The 3' adapter is trimmed first, followed by the 5' leaving only the region of interest in a fastq format.


![The 3' adapter is trimmed first, followed by the 5' leaving only the region of interest in a fastq format.](images\librarytrimming.svg)


#### **Parameters Used:**

* <span style="color:Purple">**-a / -g**</span>   -  specifies 3' or 5'regular adapters </br>
* <span style="color:Purple">**-m 6**</span>   -  minimum length, set during the 5' adapter trimming step, any read under the specified <span style="color:Purple">**length 6**</span> will be discarded. </br>
* <span style="color:Purple">**Default Maximum Error Rate**</span>   - though not specified, the maximum error rate can not exceed <span style="color:Purple">**10%**</span>

---

### **Step 4:** Quality trim and tally sequences

ROI reads are then filtered for quality using <span style="color:Purple">**fastq_quality_filter**</span> and converted to fasta format using <span style="color:Purple">**fastq_to_fasta**</span>, both tools also come from <span style="color:Purple">**FASTX_toolkit**</span>. A basic R script is run to tally unique library sequences and a tab-separated file is produced with the sequence in the first column and number of occurrences in the second.


![](images\libraryqualityfilter.svg)
#### **Parameters Used:**

* <span style="color:Purple">**-q 20**</span>   -  phred based quality score of 20 </br>
* <span style="color:Purple">**-p 100**</span>   - percent of bases that must meet phred quality score minimum is <span style="color:Purple">**100%**

---

### **Pre-Downstream Analysis Steps:** Translating seqeunces, normalizing counts to reads per million, and optional filters**

Once the main pipeline is complete, the next steps are flexible based on what the project requires. Most projects will included translation of the nucleic acid sequences to peptide sequences, normalizing reads by rpm, and a few optional library filters including filters by length, stop codon removal, oligo lists, cheater sequences, and more. 



